pyproject.toml

```diff
@@ -5,13 +5,13 @@ authors = [
     {name = "Kiarash Kiani", email = "kiarash@datachef.co"},
 ]
 dependencies = [
-    "cdktf>=0.20.8",
     "rich>=13.7.1",
-    "cdktf-cdktf-provider-aws>=19.27.0",
-    "cdktf-cdktf-provider-azurerm>=12.25.0",
     "boto3>=1.34.147",
     "psutil>=6.0.0",
     "flask>=3.0.3",
+    "pulumi>=3.127.0",
+    "pulumi-aws>=6.47.0",
+    "pulumi-azure-native>=2.51.0",
```

Comment:
I'd recommend moving it to `devenv.nix`. Use `pulumi-bin` which contains these requirements.

Reply:
@shahinism Wouldn't doing this then leave pulumi out of the dependencies when we publish damavand as a package to pypi?

Reply:
That's a very valid point. However, then we're going to have a bit of a challenge here:
1. we need to make sure the version installed here, always confirms the version on `devenv` 🤔
2. it won't cause https://github.com/pulumi/pulumi/issues/14525 for the consumers of your library.

Reply:
FYI, remember that pulumi needs to be BOTH installed as a package and as CLI.

Reply:
Thanks @shahinism. I do agree that we need to be aware of that. However, as we are not deploying actually anything to a cloud from this repo (the is the library not an application), using pdm to install the packages should heart for now. I am thinking that more down the road for more deep automated testing we might actually need to deploy things and we need to be aware of this.

I will create an issue explaining this so it won't get lost.

Reply:
@wtfzambo How is that relevant here? this is for dependencies of the package Damavand plus when the maintainers work on it and need the lsp, linter, and tests recognize pulumi python sdk.

Reply:
@kkiani I was under the impression that nix would also take care of installing the necessary binaries